tests/unit/test_util.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+"""Tests for craft_providers helper utilities."""
+from unittest import mock
+
+import pytest
+from craft_providers.util import retry
+
+
+@pytest.fixture(params=range(1, 6), ids=[f"timeout_{i}ms" for i in range(1, 6)])
+def timeout(request):
+    return request.param * 0.001
+
+
+@pytest.fixture(params=range(5), ids=[f"wait_{i}us" for i in range(5)])
+def retry_wait(request):
+    """A parametrized fixture of times to wait before retrying, in microseconds"""
+    return request.param * 0.000_001
+
+
+def test_retry_until_timeout_success(failure_count, retry_wait):
```

Comment:
Do we need to test a scenario where `retry_wait` > `timeout`?

Asking because that was the original bug that began this.